### ISSUE TITLE
GL Conflict Between this and CoFH Core

### ISSUE THREAD

**Issue by Parker8283** [OpenMods/OpenModsLib#43]
With both CoFH Core 3.0.0RC7 and OpenModsLib 0.6 installed, I receive the following error:
<pre>[22:41:54] [Client thread/ERROR]: ########## GL ERROR ##########
[22:41:54] [Client thread/ERROR]: @ Pre startup
[22:41:54] [Client thread/ERROR]: 1286: Invalid framebuffer operation</pre>
This causes nothing to draw on screen ever. This happens once when the Mojang logo is trying to be drawn, and then it gets spammed in the console when trying to draw the main menu.

Also reported here: https://github.com/CoFH/Feedback/issues/308

**Comment by boq** [comment]
Hmm, can't reproduce with CoFHCore only . So it's probably yet another non-standard graphic drivers compatibility problem.

Anyway, you can disable OML functionality with `OpenModsLibCore.json` (`activate_gl_capabilities_hook` and `activate_stencil_patches`). Please try to see what happens when you disable those (just single one or both).

**Comment by Parker8283** [comment]
Yea, I couldn't reproduce with just CoFH Core or OML. So I figured it was a mod compatibility issue. I'll try disabling them and see what happens.

**Comment by boq** [comment]
Oh, I meant I can't reproduce with CoFHCore + OML too. And, as I wrote in other CoFH issue, I'm most sure if CoFH code is even activating (though I tried to test that in dev and made sure it is called - still nothing).

**Comment by Parker8283** [comment]
This is what skyboy linked me last night in IRC, which is what they're doing. Not sure if it helps you or not, I don't OpenGL :P

https://github.com/CoFH/CoFHCore/blob/master/src/main/java/cofh/core/ProxyClient.java Line 114.

**Comment by boq** [comment]
Already found it: https://github.com/CoFH/Feedback/issues/308#issuecomment-80949073. It that's all they are doing, then it's... weird.

**Comment by Parker8283** [comment]
Disabling either of the two that you told me to fixes the issue. Though with `activate_stencil_patches` enabled, this gets logged a dozen times:
<pre>[12:30:02] [Client thread/WARN] [OpenMods]: openmods.stencil.FramebufferHooks.createRenderbufferStorage(FramebufferHooks.java:24): Packet depth+stencil buffer not supported</pre>
Hope it somewhat helps.

**Comment by boq** [comment]
This is normal, `activate_gl_capabilities_hook` is relatively harmless, but disabling it will prevent other part of code from discovering stencil buffer capabilities.

Also, can you crash you client (with F3+C)? There is some data in crashlog that might be helpful.

**Comment by Parker8283** [comment]
With everything enabled?

**Comment by Parker8283** [comment]
Hmm...it appears that that doesn't work from the main menu...but I did find this with all enabled:
<pre>[12:49:33] [Client thread/WARN] [OpenMods]: openmods.stencil.FramebufferHooks.createRenderbufferStorage(FramebufferHooks.java:22): Your potato failed to allocate nice buffer. No stencils for you. Cause: Invalid enum
...
[12:49:33] [Client thread/WARN] [OpenMods]: openmods.stencil.FramebufferHooks.createRenderbufferStorage(FramebufferHooks.java:22): Your potato failed to allocate nice buffer. No stencils for you. Cause: Invalid framebuffer operation</pre>

**Comment by boq** [comment]
Ok, I did debugging mod: https://dl.dropboxusercontent.com/u/7088748/gltrace-1.0.jar. It will collect as much as it can and then die. Please run and post all logs (there will be quite a log of it).

**Comment by Parker8283** [comment]
Well this is interesting. The Mojang logo was able to be drawn with this mod installed. Also, love the Runtime Exception message :P

http://paste.ee/p/zSdtp

**Comment by boq** [comment]
Well, it works differently because I derped and actually changed behaviour of one part of code. Anyway, got interesting results in there (bad call in CoFH Core, commented in parallel thread).

Also, update you computer :P Really, even my 7 years old potato supports GL30.

**Comment by Parker8283** [comment]
I would, but, you know, Apple. u.u

**Comment by boq** [comment]
Ok, so it seems that all currently known bugs on CoFH side are fixed, but there is probably problem on my part too.

So rerun plz -> https://dl.dropboxusercontent.com/u/7088748/gltrace-1.1.jar (I removed that side effect, should give more info)

**Comment by Parker8283** [comment]
I self-compiled CoFH Core with their fixes, and this was the result from the new gltrace: http://paste.ee/r/EMAxL Also, without gltrace, the game runs just fine.

Just in case, I then downgraded back to the version I was using before and ran with the new gltrace: http://paste.ee/r/Mj9ba This time, the Mojang logo wasn't drawn.

Hope it helps. Also, I closed the CoFH Core issue because the original issue itself has been fixed. Technically, the whole reason I opened both of these issues has been fixed, but it'd be good to fix it completely I guess.
